src/models/connectionStore.ts

```diff
@@ -131,8 +131,8 @@ export class ConnectionStore {
                 // Don't try to lookup a saved password if savePassword is set to false for the credential
                 resolve(credentialsItem);
             } else if (ConnectionCredentials.isPasswordBasedCredential(credentialsItem.connectionCreds)
-                    && Utils.isEmpty(credentialsItem.connectionCreds.password)) {
-
+                    && Utils.isEmpty(credentialsItem.connectionCreds.password)
+                    && credentialsItem.connectionCreds['emptyPasswordInput'] === false) {
```

Comment:
This causes it to only save if emptyPasswordInput is explicitly false. Since old configs won't have a value for emptyPasswordInput, I think it would be safer to instead check if emptyPasswordInput is not explicitly true, that way undefined is treated as false.

Since this is typescript and not javascript I'd also suggest looking into a more type safe way to do this, or at least explaining in a comment that emptyPasswordInput will be present only when connectionCreds is actually a ConnectionProfile object.